src/test/java/com/example/webapp/Integration/DepartmentControllerIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+package com.example.webapp.Integration;
+
+import com.example.webapp.Exception.DepartmentNotFoundException;
+import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.Service.DepartmentService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.flywaydb.core.Flyway;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Objects;
+
+import static com.example.webapp.util.DEPARTMENT_LIST;
+import static com.example.webapp.util.INVALID_DEPARTMENT_DTO;
+import static com.example.webapp.util.NEW_DEPARTMENT_DTO;
+import static com.example.webapp.util.REVERSE_DEPARTMENT_LIST_DTO;
+import static com.example.webapp.util.VALID_DEPARTMENT;
+import static com.example.webapp.util.VALID_DEPARTMENT_DTO;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:application-test.properties")
+@Transactional
+class DepartmentControllerIntegrationTest {
+
+    @Autowired
+    private DepartmentRepository departmentRepository;
+
+    @Autowired
+    Flyway flyway;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private DepartmentService departmentService;
+
+    @BeforeEach
+    public void cleanUp() {
+        flyway.clean();
+        flyway.migrate();
+    }
+
+    @Test
+    void getAll_shouldReturnAllDepartments() throws Exception {
+        departmentRepository.saveAll(DEPARTMENT_LIST);
+        MvcResult mvcResult = mockMvc.perform(get("/departments")
```

Comment:
Good to know: For integration test Srpring Boot also provide TestRestTemplate and the test will be performed on application that will run in a container